workflow/controller/pod_cleanup.go

```diff
@@ -23,7 +25,12 @@ func (woc *wfOperationCtx) queuePodsForCleanup() {
 			continue
 		}
 		nodeID := woc.nodeID(pod)
-		if !woc.wf.Status.Nodes[nodeID].Phase.Fulfilled() {
+		nodePhase, err := woc.wf.Status.Nodes.GetPhase(nodeID)
+		if err != nil {
+			woc.log.Fatalf("was unable to obtain node for %s", nodeID)
+			panic(fmt.Sprintf("node was not present with id %s", nodeID))
```

Comment:
I'd suggest we shouldn't abandon looking through all the other pods here, and should complain loudly but `continue`